pkg/internal/storage/repo/syncrepo/find_chunk.go

```diff
@@ -0,0 +1,49 @@
+package syncrepo
+
+import (
+	"context"
+	"fmt"
+
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type gormScope = func(*gorm.DB) *gorm.DB
+
+func findChunk[Model, ResultModel any](
+	ctx context.Context,
+	db *gorm.DB,
+	tableName, stringID string,
+	scopes []gormScope,
+) ([]*ResultModel, error) {
+	var resultModels []*ResultModel
+
+	var model Model
```

Comment:
```suggestion
	modelPtr := new(Model)
```
then later `Model(modelPtr)` would not be safer ?

Reply:
This doesn't change anything. 
Is it safer?: `no`

Reply:
sure `modelPtr := new(Model)` can make the intent clearer to me .. but ok in behave i agree with u